src/spi_bus.cpp

```diff
@@ -0,0 +1,392 @@
+}
+    
+void SpiBus::set_command(const size_t udriver_id, 
+                         const MotorBoardCommand& command)
+{
+    command_[udriver_id]->append(command);
+}
+
+void SpiBus::send_if_input_changed()
+{
+    // send command if a new one has been set ----------------------------------
+    bool commands_have_changed = false;
+    for(size_t i = 0 ; i < nb_udrivers_ ; ++i)
+    {
+        if(command_[i]->has_changed_since_tag())
+        {
+            commands_have_changed = true;
+            break;
+        }
+    }
+    if(commands_have_changed)
+    {
+        send_newest_command();
+    }
+    // send controls if a new one has been set ---------------------------------
+    bool controls_have_changed = false;
+    for(size_t i = 0 ; i < control_.size() ; ++i)
+    {
+        if(control_[i]->has_changed_since_tag())
+        {
+            controls_have_changed = true;
+            break;
+        }
+    }
+    if(controls_have_changed)
+    {
+        send_newest_controls();
+    }
+}
+
+void SpiBus::send_newest_command()
+{
+    for(size_t i = 0 ; i < nb_udrivers_ ; ++i)
+    {
+        MotorInterface::ScalarTimeseries::Index timeindex =
+            command_[i]->newest_timeindex();
+        MotorBoardCommand command = (*command_[i])[timeindex];
+        command_[i]->tag(timeindex);
+        sent_command_[i]->append(command);
+
+        // Some aliases
+        MotorDriver& udriver = main_board_interface_->motor_drivers[i];
+        const uint32_t& command_id = command_[i]->newest_element().id_;
+        const int32_t& command_content = command_[i]->newest_element().content_;
+        switch (command_id)
+        {
+            case MotorBoardCommand::ENABLE_SYS:
+                command_content == MotorBoardCommand::ENABLE ?
+                    udriver.Enable() : udriver.Disable();
+                break;
+            case MotorBoardCommand::ENABLE_MTR1:
+                command_content == MotorBoardCommand::ENABLE ?
+                    udriver.motor1->Enable() : udriver.motor1->Disable();
+                break;
+            case MotorBoardCommand::ENABLE_MTR2:
+                command_content == MotorBoardCommand::ENABLE ?
+                    udriver.motor2->Enable() : udriver.motor2->Disable();
+                break;
+            case MotorBoardCommand::ENABLE_VSPRING1:
+                break;
+            case MotorBoardCommand::ENABLE_VSPRING2:
+                break;
+            case MotorBoardCommand::SEND_CURRENT:
+                break;
+            case MotorBoardCommand::SEND_POSITION:
+                break;
+            case MotorBoardCommand::SEND_VELOCITY:
+                break;
+            case MotorBoardCommand::SEND_ADC6:
+                break;
+            case MotorBoardCommand::SEND_ENC_INDEX:
+                break;
+            case MotorBoardCommand::SEND_ALL:
+                break;
+            case MotorBoardCommand::SET_CAN_RECV_TIMEOUT:
+                break;
```

Comment:
Are these commands not supported by SPI? If yes, it might be good to add a comment saying this.  Maybe issuing a warning when trying to send an unsupported command would be good as well.